cmd/node-joiner/nodejoiner_integration_test.go

```diff
@@ -0,0 +1,230 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"go/build"
+	"os"
+	"path"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/rogpeppe/go-internal/testscript"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"sigs.k8s.io/controller-runtime/pkg/envtest"
+	"sigs.k8s.io/yaml"
+
+	v1 "github.com/openshift/api/config/v1"
+	"github.com/openshift/installer/pkg/clusterapi"
+)
+
+const (
+	testResourcesFolder = "setup"
+)
+
+func TestMain(m *testing.M) {
+
+	os.Exit(testscript.RunMain(m, map[string]func() int{
+		"node-joiner": func() int {
+			if err := nodeJoiner(); err != nil {
+				return 1
+			}
+			return 0
+		},
+	}))
+}
+
+func TestNodeJoinerIntegration(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping integration test")
+	}
+
+	projectDir, err := os.Getwd()
+	assert.NoError(t, err)
+
+	// Prepare the temporary cluster that will be used by node-joiner.
+	testEnv := &envtest.Environment{
+		CRDDirectoryPaths: []string{
+			// Preload OpenShift specific CRDs so that it will be possible to create specific OpenShift resources.
+			filepath.Join(build.Default.GOPATH, "pkg", "mod", "github.com", "openshift", "api@v0.0.0-20240301093301-ce10821dc999", "config", "v1"),
```

Comment:
Was seeing:
```
[rwsu@installer]$ go test ./cmd/node-joiner/.
--- FAIL: TestNodeJoinerIntegration (0.07s)
    --- FAIL: TestNodeJoinerIntegration/add-nodes (5.58s)
        testscript.go:534: FAIL: testdata/add-nodes.txt:0: unable to install CRDs onto control plane: unable to read CRD files: stat /home/rwsu/go/pkg/mod/github.com/openshift/api@v0.0.0-20240301093301-ce10821dc999/config/v1: no such file or directory
```
If you having troubling running the new integration test, I had to do the following:
```
export OPENSHIFT_INSTALL_CLUSTER_API=YES
./hack/build.sh
go mod vendor
go mod tidy
go test ./cmd/node-joiner/.
```